### DISCUSSION TITLE
Geant4 CMake Error

### DISCUSSION THREAD

**Discussion by bpnuke** [Homebrew/discussions#1732] [Writing Formulae/Casks]
```
cmake -DGEANT4_BUILD_MULTITHREADED=ON        \
      -DGEANT4_USE_QT=ON                     \
      -DGEANT4_USE_GDML=ON                   \
      -DGEANT4_USE_OPENGL_X11=ON             \
      -DGEANT4_USE_RAYTRACER_X11=ON          \
      -DGEANT4_USE_SYSTEM_ZLIB=ON            \
      -DGEANT4_BUILD_CXXSTD=14               \
      -DGEANT4_INSTALL_DATA=ON               \
      -DGEANT4_INSTALL_DATA_TIMEOUT=86400    \
      -DCMAKE_INSTALL_PREFIX=../g4-install/. \
      -DCMAKE_CXX_COMPILER=clang++           \
      ../g4-git/.
cmake --build . -j16
cmake --build . --target install
```

_On calling the above build commend for geant4 from master in g4-git from g4-build, I receive the following error. I believe there may be an issue with the homebrew qt@5 binary_

CMake Error at /usr/local/lib/cmake/Qt5Core/Qt5CoreConfig.cmake:14 (message):
  The imported target "Qt5::Core" references the file

     "/usr/local/.//mkspecs/macx-clang"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/usr/local/lib/cmake/Qt5Core/Qt5CoreConfigExtras.cmake"

  but not all the files it references.
```
Call Stack (most recent call first):
  /usr/local/lib/cmake/Qt5Core/Qt5CoreConfigExtras.cmake:56 (_qt5_Core_check_file_exists)
  /usr/local/lib/cmake/Qt5Core/Qt5CoreConfig.cmake:227 (include)
  cmake/Modules/G4InterfaceOptions.cmake:140 (find_package)
  cmake/Modules/G4CMakeMain.cmake:64 (include)
  CMakeLists.txt:51 (include)
```

-- Configuring incomplete, errors occurred!
See also "/Users/d3x381/Applications/g4-build/CMakeFiles/CMakeOutput.log".

**Comment by cho-m** [comment]
Based on paths, I think you force linked the keg-only `qt@5` formula.

It looks like the problem is caused by some relative paths in Qt5's .cmake files, which are expecting the original unlinked locations and by using them from linked directory `/usr/local/lib/cmake`, CMake is unable to find the correct files/directories.

The way to get this to work right now would be to add the necessary compiler flags as specified in:
```console
❯ brew info qt@5
...
If you need to have qt@5 first in your PATH, run:
  echo 'export PATH="/usr/local/opt/qt@5/bin:$PATH"' >> ~/.zshrc

For compilers to find qt@5 you may need to set:
  export LDFLAGS="-L/usr/local/opt/qt@5/lib"
  export CPPFLAGS="-I/usr/local/opt/qt@5/include"

For pkg-config to find qt@5 you may need to set:
  export PKG_CONFIG_PATH="/usr/local/opt/qt@5/lib/pkgconfig"
```

---

For a more detailed look at issue:

- `get_filename_component(_qt5Core_install_prefix "${CMAKE_CURRENT_LIST_DIR}/../../../" ABSOLUTE)`
  - Using recommended path: `/usr/local/opt/qt@5/lib/cmake/Qt5Core/Qt5CoreConfig.cmake` gets `/usr/local/opt/qt@5`
  - Using force-linked path: `/usr/local/lib/cmake/Qt5Core/Qt5CoreConfig.cmake` gets `/usr/local`
- `set(_qt5_corelib_extra_includes "${_qt5Core_install_prefix}/.//mkspecs/macx-clang")`
  - Using recommended path results in `/usr/local/opt/qt@5/mkspecs/macx-clang`
  - Using force-linked path results in `/usr/local/mkspecs/macx-clang`

You can see that the default installed (keg-only/unlinked) path will find the correct directory while the force-linked path is non-existent.

---

Probably needs some more digging into to see if there is a way to modify formula to allow force linkage to work correctly. As `qt` (v6) formula should work from linked path, there may by some way to deal with this.

**Reply by bpnuke** [reply]
That was exactly the issue THANK YOU!
